openstf/pipeline/train_model.py

```diff
@@ -159,25 +159,26 @@ def train_model_pipeline_core(
     logging.info("Fitted a new model, not yet stored")
 
     # Check if new model is better than old model
-    if old_model is not None:
+    if old_model:
         combined = train_data.append(validation_data)
         x_data, y_data = combined.iloc[:, 1:], combined.iloc[:, 0]
 
         # Score method always returns R^2
         score_new_model = model.score(x_data, y_data)
-        score_old_model = old_model.score(x_data, y_data)
+        try:
+            score_old_model = old_model.score(x_data, y_data)
```

Comment:
Just being curious: Why is the try, catch  required here?

Reply:
if the old model has different feature names compared to the new model, I'll add it as a comment